app/controllers/shift_surveys_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ShiftSurveysController < ApplicationController
+  skip_before_action :authenticate_user!
+
+  def show
+    @shift_survey = ShiftSurvey.find_by(token: params[:token])
+    if @shift_survey.nil?
+      flash[:notice] = 'Invalid survey token'
+      redirect_to root_path
+      return
+    end
+    
+    if @shift_survey.status == "Complete"
+      flash[:notice] = 'You have already submitted your survey'
+      render 'thanks'
+    end
+  end
+
+  def update
+    @shift_survey = ShiftSurvey.find_by(token: params[:token])
+    ["supplies", "response_time", "hours_match"].each do |field|
+      if params[:shift_survey]["#{field}"] === "true"
+        params[:shift_survey]["#{field}_text"] = ""
+      end
+    end
```

Comment:
Per our zoom discussion - if the inputs are disabled when hidden, their values will not be submitted and I think you'd be able to remove this code.